### PR TITLE
feat(influx publisher): allow drivers to define measurement names

### PR DESCRIPTION
**Pull request by stakach**
this mostly effects the more complex measurements being taken so a single driver like AreaManagement can output wireless, desk and space measurements - making it easier to query the data.

### CODE REVIEW

**Comment by stakach** [comment]
Based on measurement examples in the influx documentation
https://docs.influxdata.com/influxdb/v1.8/concepts/schema_and_data_layout/#avoid-encoding-data-in-measurements-and-keys

For simple status keys the existing method of putting data in is sufficient.
i.e. the Bookings driver emits `booked`, `in-use`, `pending` and `presence` etc those fields going in under a measurement called Booking is fine

But for things like AreaManagement we want to split up the measurements to make it simple to query

**Comment by stakach** [comment]
The spec failures look to be related to password generation on the partner env
@w-le is that something you can look at?
image

File: src/source/publishing/influx_publisher.cr [hunk 1]
```diff
@@ -79,6 +82,7 @@ module PlaceOS::Source
       end
       tags["pos_system"] = data.control_system_id
       tags["pos_index"] = data.index.to_i64.to_s
+      tags["mod_name"] = data.module_name
```
Comment (caspiano) [inline src/source/publishing/influx_publisher.cr:85]: Should we keep our fields `pos_` prefixed for consistency?

Comment (stakach) [reply src/source/publishing/influx_publisher.cr:85]: yeah good idea

**Comment by caspiano** [comment]
@stakach I've resolved that in the latest `init` image

**Comment by stakach** [comment]
nice, how do we get it to use that tag?

File: src/source/publishing/influx_publisher.cr [hunk 2]
```diff
@@ -144,44 +154,54 @@ module PlaceOS::Source
...
+
+      points
+    end
+
+    protected def self.build_custom_point(measurement, data, fields, local_tags, compacted, timestamp, ts_map, ts_tag_keys)
+      # Add the fields
+      local_fields = fields.dup
+      compacted.each do |sub_key, value|
+        sub_key = (ts_map[sub_key]? || sub_key).gsub(/\W/, '_')
+        if sub_key == "measurement" && value.is_a?(String)
```
Comment (caspiano) [inline src/source/publishing/influx_publisher.cr:181]: `pos_measurement`?

Comment (stakach) [reply src/source/publishing/influx_publisher.cr:181]: this one is on the driver side technically, we might want to call it `measurement_name` but I think `measurement` is fine

**Pull request merged**